src/main/java/org/cactoos/iterable/Paged.java

```diff
@@ -46,14 +42,17 @@
      * <p>
      * @param first First bag of elements
      * @param next Subsequent bags of elements
-     * @param <I> Custom iterator
      */
-    public <I extends Iterator<X>> Paged(
-        final I first, final Func<I, I> next
+    public Paged(
+        final Iterable<? extends X> first,
+        final Func<? super Iterable<? extends X>, ? extends Iterable<? extends X>> next
     ) {
         super(
             new IterableOf<>(
-                new org.cactoos.iterator.Paged<>(first, next)
+                new org.cactoos.iterator.Paged<X>(
```

Comment:
@SergiusAC It's better to use `Scalar<Iterator>` here (see #1592)

Reply:
Done it.